### DISCUSSION TITLE
Error when calling rpc from within edge function

### DISCUSSION THREAD

**Discussion by magick93** [supabase/supabase#14169] [General]
Hello

I'm trying to run edge functions locally. The function is deloying, and I can call it locally, and it is working as expected but for when I try to insert data using a posgres function / rpc call.

I get the following error:

```
FetchError: error sending request for url (http://localhost:54321/rest/v1/rpc/insert_items_dynamic): error trying to connect: tcp connect error: Cannot assign requested address (os error 99)
```

I'm wondering if there is a port conflict because the supabase api and functions are using the same port.

**Comment by rienheuver** [comment]
I'm having this issue as well. I'm not calling an RPC though, I'm just performing a select query with supabase-js:
```
TypeError: error sending request for url (http://localhost:54321/rest/v1/profiles?select=id&email=eq.piet%40example.com): error trying to connect: tcp connect error: Cannot assign requested address (os error 99)
```

**Comment by rienheuver** [comment]
Apparently this was asked in the supabase discord last month and they figured it out. I still believe this should be in the docs somewhere, because this is highly confusing.
Here's the discord thread: https://discord.com/channels/839993398554656828/1110601954704490537

The solution is:
```
Oh nevermind - I figured it out. For anyone else that's trying to do this, use this as your supabase url:

http://host.docker.internal:54321/ (rather than localhost:54321)
```

**Reply by northfacejmb** [reply]
thank you so much for this

**Reply by Seanmclem** [reply]
all my deno edge function issues suddenly gone. Thanks!

**Reply by yuya1219** [reply]
My hard problem has solved!!
Thank you berry match!! ^o^

**Reply by okumujustine** [reply]
Saved me many hours, thank you

**Comment by gregnr** [comment]
Hey guys! Best practice when connecting to Supabase API from within Edge Functions is to use the built-in `SUPABASE_URL` environment variable. This will automatically resolve to the correct Supabase project API both locally and when deployed to the cloud. 

More info here:
https://supabase.com/docs/guides/functions/secrets#default-secrets

**Reply by rienheuver** [reply]
Thanks, that makes a lot more sense :+1:

**Comment by AdamEisfeld** [comment]
I have this issue as well. I am serving 2 edge functions locally, and I want one edge function to forward a request to the other (one edge function is guarded behind JWT verification, the other is a sort of gateway that allows requests without JWTs through to whitelisted endpoints in the first function).

When I attempt to forward the request from the first function to the second, I receive the "tcp connect error: Cannot assign requested address (os error 99)" error.

I've tried switching from using the built-in SUPABASE_URL env var to "http://host.docker.internal:54321/functions/v1/..." to no avail.
